test/inference_test.jl

```diff
@@ -1026,3 +1026,40 @@ end
 
     @test all(result.predictions[:y] .== Bernoulli(mean(Beta(1.0, 1.0))))
 end
+
+@testitem "Test misspecified types in infer function" begin
+    @model function rolling_die(n)
+        y = datavar(Vector{Float64}, n)
+
+        θ ~ Dirichlet(ones(6))
+        for i in 1:n
+            y[i] ~ Categorical(θ)
+        end
+    end
+
+    observations = [[1.0; zeros(5)], [zeros(5); 1.0]]
+
+    @testset "Test misspecified data" begin
+        @test_throws ErrorException infer(model = rolling_die(2), data = (y = observations))
```

Comment:
These tests don't really test that the output error is user friendly. It will error of course anyway no matter the changes in the PR. I suggest to use 
```
@test_throws "part of the error message" infer(...)
``` 
See example [here](https://github.com/ReactiveBayes/GraphPPL.jl/blob/dev-4.0.0/test/model_generator_tests.jl#L28)

Reply:
done